lib/src/Base/Stat/IsotropicCovarianceModel.cxx

```diff
@@ -0,0 +1,270 @@
+  }
+  outputDimension_ = kernelOutputDimension;
+  kernel_ = oneDimensional;
+}
+
+CovarianceModel IsotropicCovarianceModel::getKernel() const
+{
+  return kernel_;
+}
+
+/* Virtual constructor */
+IsotropicCovarianceModel * IsotropicCovarianceModel::clone() const
+{
+  return new IsotropicCovarianceModel(*this);
+}
+
+/* Computation of the covariance function */
+SquareMatrix IsotropicCovarianceModel::operator() (const Point & s,
+  const Point & t) const
+{
+  return (*this)(t - s);
+}
+
+SquareMatrix IsotropicCovarianceModel::operator() (const Point & tau) const
+{
+  // for the time being, the output dimension is necessarily 1.
+  SquareMatrix result(1);
+  result(0, 0) = computeAsScalar(tau);
+  return result;
+}
+
+/* Computation of the covariance function */
+Scalar IsotropicCovarianceModel::computeAsScalar(const Point & tau) const
+{
+  if (tau.getDimension() != inputDimension_) throw InvalidArgumentException(HERE) << "Error: expected a shift of dimension=" << inputDimension_ << ", got dimension=" << tau.getDimension();
+  const Point tauNorm = Point(1, tau.norm());
+  return kernel_.computeAsScalar(Point(1), tauNorm);
+}
+
+
+Scalar IsotropicCovarianceModel::computeAsScalar(const Point & s,
+  const Point & t) const
+{
+  if (s.getDimension() != inputDimension_) throw InvalidArgumentException(HERE) << "Error: the point s has dimension=" << s.getDimension() << ", expected dimension=" << inputDimension_;
+  if (t.getDimension() != inputDimension_) throw InvalidArgumentException(HERE) << "Error: the point t has dimension=" << t.getDimension() << ", expected dimension=" << inputDimension_;
+  const Point tau = t - s;
+  return computeAsScalar(tau);
+}
+
+Scalar IsotropicCovarianceModel::computeAsScalar(const Collection<Scalar>::const_iterator & s_begin,
+  const Collection<Scalar>::const_iterator & t_begin) const
+{
+  Scalar tauNormSquare = 0;
+  Collection<Scalar>::const_iterator s_it = s_begin;
+  Collection<Scalar>::const_iterator t_it = t_begin;
+  for (UnsignedInteger i = 0; i < inputDimension_; ++i, ++s_it, ++t_it)
+  {
+    const Scalar dx = *s_it - *t_it;
+    tauNormSquare += dx * dx;
+  }
+  const Point tauNorm = Point(1, std::sqrt(tauNormSquare));
+  return kernel_.computeAsScalar(Point(1), tauNorm);
+}
+
+/* Gradient */
+Matrix IsotropicCovarianceModel::partialGradient(const Point & s,
+  const Point & t) const
+{
+  if (s.getDimension() != inputDimension_) throw InvalidArgumentException(HERE) << "Error: the point s has dimension=" << s.getDimension() << ", expected dimension=" << inputDimension_;
+  if (t.getDimension() != inputDimension_) throw InvalidArgumentException(HERE) << "Error: the point t has dimension=" << t.getDimension() << ", expected dimension=" << inputDimension_;
+  const Point tau = t - s;
+  const Scalar tauNorm = tau.norm();
+  const Point normGradient(tau / tauNorm);
```

Comment:
and here

Reply:
I don't understand this comment as I did use the constructor for the complex type Point:
`   const Point normGradient(tau / tauNorm); `

Reply:
It was a mouse slip. See line 145.